src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/ComponentParameterNullableWarningSuppressorTests.cs

```diff
@@ -150,27 +148,129 @@ public class EditorRequiredAttribute : Attribute { }
+        {
+            var testCode = """
+                #nullable enable
+                using System;
+                using Microsoft.AspNetCore.Components;
+                
+                public class MyComponent
+                {
+                    [Parameter, EditorRequired]
+                    public string MyParameter { get; set; }
+                }
+
+                namespace Microsoft.AspNetCore.Components
+                {
+                    public class ParameterAttribute : Attribute { }
+                    public class EditorRequiredAttribute : Attribute { }
+                }
+
+                """;
+
+            await VerifyAnalyzerAsync(testCode,
+                    // /0/Test0.cs(8,19): warning CS8618: Non-nullable property 'MyParameter' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+                    DiagnosticResult.CompilerWarning("CS8618").WithSpan(8, 19, 8, 30).WithSpan(8, 19, 8, 30).WithArguments("property", "MyParameter").WithIsSuppressed(true),
+                    // /0/Test0.cs(7,6): warning CS0436: The type 'ParameterAttribute' in '/0/Test0.cs' conflicts with the imported type 'ParameterAttribute' in 'Microsoft.AspNetCore.Components, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. Using the type defined in '/0/Test0.cs'.
+                    DiagnosticResult.CompilerWarning("CS0436").WithSpan(7, 6, 7, 15).WithArguments("/0/Test0.cs", "Microsoft.AspNetCore.Components.ParameterAttribute", "Microsoft.AspNetCore.Components, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60", "Microsoft.AspNetCore.Components.ParameterAttribute"),
+                    // /0/Test0.cs(7,17): warning CS0436: The type 'EditorRequiredAttribute' in '/0/Test0.cs' conflicts with the imported type 'EditorRequiredAttribute' in 'Microsoft.AspNetCore.Components, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. Using the type defined in '/0/Test0.cs'.
+                    DiagnosticResult.CompilerWarning("CS0436").WithSpan(7, 17, 7, 31).WithArguments("/0/Test0.cs", "Microsoft.AspNetCore.Components.EditorRequiredAttribute", "Microsoft.AspNetCore.Components, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60", "Microsoft.AspNetCore.Components.EditorRequiredAttribute")
+                );
+        }
+
+        [Theory]
+        [InlineData("internal")]
+        [InlineData("private")]
+        [InlineData("protected internal")]
+        [InlineData("protected")]
+        [InlineData("public static")]
+        public async Task IncorrectModifiersStillReport(string modifiers)
+        {
+            var testCode = $$"""
+                #nullable enable
+                using System;
+                using Microsoft.AspNetCore.Components;
+
+                public class MyComponent : ComponentBase
+                {
+                    [Parameter]
```

Comment:
Did you mean test EditorRequired? Otherwise it doesn't seem like an interesting test since without EditorRequied the analyzer always bails.

```suggestion
                    [Parameter, EditorRequired]
```